src/ConnectedMode/SlCoreConnectionAdapter.cs

```diff
@@ -202,22 +205,33 @@ private bool TryGetConnectionConfigurationSlCoreService(out IConnectionConfigura
         logger.LogVerbose($"[{nameof(IConnectionConfigurationSLCoreService)}] {SLCoreStrings.ServiceProviderNotInitialized}");
         return false;
     }
-
-    private static ValidateConnectionParams GetValidateConnectionParams(ConnectionInfo connectionInfo, Either<TokenDto, UsernamePasswordDto> credentials)
-    {
-        return new ValidateConnectionParams(GetTransientConnectionDto(connectionInfo, credentials));
-    }
-
-    private static GetAllProjectsParams GetAllProjectsParams(ConnectionInfo connectionInfo, Either<TokenDto, UsernamePasswordDto> credentials)
+    
+    private static Either<TransientSonarQubeConnectionDto, TransientSonarCloudConnectionDto> GetTransientConnectionDto(ConnectionInfo connectionInfo, ICredentials credentials)
     {
-        return new GetAllProjectsParams(GetTransientConnectionDto(connectionInfo, credentials));
+        var credentialsDto = MapCredentials(credentials);
+        
+        return connectionInfo.ServerType switch
+        {
+            ConnectionServerType.SonarQube => Either<TransientSonarQubeConnectionDto, TransientSonarCloudConnectionDto>.CreateLeft(
```

Comment:
I agree that it would make it a bit more clear. However, I am not sure whether this is the correct place. This requirement comes from the SLCore API, which is already documented there, and on our end, we have unit tests to verify that this behavior is respected.